Sources/Momento/messages/responses/topics/TopicSubscribe.swift

```diff
@@ -58,6 +58,12 @@ public class TopicSubscription {
                 } else if (err.code == .resourceExhausted) {
                     self.logger.debug("Too many subscribers, not resubscribing")
                     return nil
+                } else if (err.code == .unavailable) {
+                    // If the topic client is closed but the subscription is still open, we'll
+                    // receive a "shutdown" error that gets converted into an "unavailable"
+                    // GRPCStatus object. See: https://github.com/grpc/grpc-swift/blob/53e2739912b0d3090cfa6a8345fcadbe6fe2ba1a/Sources/GRPC/ConnectionPool/ConnectionPool.swift#L1025
+                    print("Connection was shutdown, not resubscribing")
+                    return nil
```

Comment:
found a fairly easy fix to catch when the GRPC connection becomes unavailable and to stop trying to resubscribe. 

tested locally and got "Connection was shutdown, not resubscribing" instead of an unending stream of `error TopicSubscribeResponse : [Momento] Received error instead of new subscription: [UNKNOWN_ERROR] Unknown error has occurred: unknown subscribe error shutdown`